dm/master/server.go

```diff
@@ -2027,3 +2037,128 @@ func (s *Server) workerArgsExtractor(args ...interface{}) (workerrpc.Client, str
 
 	return cli, worker, nil
 }
+
+// redirect gRPC request to Leader if needed
+type redirectLeaderServer struct {
+	server pb.MasterServer
+	// return true if we should redirect to master leader
+	checkFunc func() bool
+}
+
+func (r *redirectLeaderServer) StartTask(ctx context.Context, req *pb.StartTaskRequest) (*pb.StartTaskResponse, error) {
+	if r.checkFunc() {
+		return common.MasterClient().StartTask(ctx, req)
```

Comment:
We may know who is the leader before creating the client.

BTW, is it possible when _A_ is the leader and _B_, _C_ is the follower, then:
_dmctl --> B --> C --> B --> C --> ***_ and never forward the request to _A_.

Reply:
the MasterClient() is connected to  current leader node, so it's true there maybe a change requests will be forwarded to outdated leader if several node's master connect is a circle (eg A->B, B->C, C->A), but this won't last long as within one lease, nodes should all get leadership change event thus will reset connect to right leader.

In Contrast, PD add a `redirect` header to http request if request should be redirected, and if the header already exist, it just return an error. To implement this need to change the gRPC request struct, so it's not easy to do so.

Reply:
How do we ensure `the MasterClient() is connected to current leader node`?

Reply:
in election.campaign, after each leader change event, we will reconnect to new leader, so we can ensure we always connected to current leader